sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/NonStreamingOrderByDocumentQueryExecutionContext.java

```
@@ -0,0 +1,268 @@
+        this.clientSideRequestStatistics = ConcurrentHashMap.newKeySet();
+        this.maxPageSizePerPartition = maxPageSizePerPartition;
+    }
+
+    public static Flux<IDocumentQueryExecutionComponent<Document>> createAsync(
+        DiagnosticsClientContext diagnosticsClientContext,
+        IDocumentQueryClient client,
+        PipelinedDocumentQueryParams<Document> initParams,
+        DocumentCollection collection,
+        int maxPageSizePerPartition) {
+
+        QueryInfo queryInfo = initParams.getQueryInfo();
+
+        NonStreamingOrderByDocumentQueryExecutionContext context = new NonStreamingOrderByDocumentQueryExecutionContext(
+            diagnosticsClientContext,
+            client,
+            initParams.getResourceTypeEnum(),
+            initParams.getQuery(),
+            initParams.getCosmosQueryRequestOptions(),
+            initParams.getResourceLink(),
+            initParams.getQueryInfo().getRewrittenQuery(),
+            new OrderbyRowComparer<>(queryInfo.getOrderBy()),
+            initParams.getCorrelatedActivityId(),
+            queryInfo.hasSelectValue(),
+            initParams.isQueryCancelledOnTimeout(),
+            maxPageSizePerPartition);
+
+        context.setTop(initParams.getTop());
+
+        try {
+            context.initialize(
+                initParams.getFeedRanges(),
+                initParams.getQueryInfo().getOrderBy(),
+                initParams.getQueryInfo().getOrderByExpressions(),
+                initParams.getInitialPageSize(),
+                collection);
+
+            return Flux.just(context);
+        } catch (CosmosException dce) {
+            return Flux.error(dce);
+        }
+    }
+
+    private void initialize(
+        List<FeedRangeEpkImpl> feedRanges, List<SortOrder> sortOrders,
+        Collection<String> orderByExpressions,
+        int initialPageSize,
+        DocumentCollection collection) throws CosmosException {
+        // Since the continuation token will always be null,
```

if there is continuationToken being passed in, should we throw exception instead of silently initialize with null?

---

Added the check in `PipelinedDocumentQueryExecutionContext` class